frontend/src/Frontend/UI/Dialogs/Receive.hs

```diff
@@ -144,56 +145,58 @@ uiReceiveModal0
      )
   => model
   -> Account key
+  -> Maybe ChainId
   -> Event t ()
   -> Workflow t m (mConf, Event t ())
-uiReceiveModal0 model account onClose = Workflow $ do
+uiReceiveModal0 model account mchain onClose = Workflow $ do
   let
-    chain = _account_chainId account
-
     netInfo = do
       nodes <- model ^. network_selectedNodes
       meta <- model ^. network_meta
       let networkId = hush . mkNetworkName . nodeVersion <=< headMay $ rights nodes
       pure $ (nodes, meta, ) <$> networkId
 
-    address = textKadenaAddress $ accountToKadenaAddress account
-
     displayText lbl v cls =
       let
         attrFn cfg = uiInputElement $ cfg
           & initialAttributes <>~ ("disabled" =: "true" <> "class" =: (" " <> cls))
       in
         mkLabeledInputView True lbl attrFn $ pure v
 
-  (showingAddr, (conf, ttl, gaslimit, transferInfo)) <- divClass "modal__main account-details" $ do
+  (showingAddr, chain, (conf, ttl, gaslimit, transferInfo)) <- divClass "modal__main receive" $ do
     rec
       showingKadenaAddress <- toggle True $ onAddrClick <> onReceiClick
 
-      (onAddrClick, _) <- controlledAccordionItem showingKadenaAddress mempty (text "Option 1: Copy and share Kadena Address")
+      elClass "h2" "heading heading_type_h2" $ text "Destination"
+      chain <- divClass "group" $ do
+        -- Network
+        void $ mkLabeledClsInput True "Network" $ \_ -> do
+          stat <- queryNetworkStatus (model ^. network_networks) $ pure $ _account_network account
```

Comment:
querying for the network status was removed in #338 and there was a function added for consistent modal headings. Might be worth rebasing `develop` ? There'll be conflicts, I imagine.